src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsVocabularyAnnotation.cs

```diff
@@ -127,8 +136,21 @@ protected IEdmTerm ComputeTerm()
         private IEdmExpression ComputeValue()
         {
             IEdmTypeReference termType = Term is UnresolvedVocabularyTerm ? null : Term.Type;
-            CsdlExpressionBase adjustedExpression = AdjustStringConstantUsingTermType((this.Annotation).Expression, termType);
 
+            if (this.Annotation.Expression == null && termType != null)
+            {
+                CsdlExpressionBase defaultExpBase = EdmTermExtensions.BuildCsdlExpression(termType, Term.DefaultValue, (CsdlLocation) this.Annotation.Location);
+
+                if (defaultExpBase == null)
+                {
+                    throw Error.NotSupported();

```

Comment:
Can we throw a better/more descriptive exception here?

Reply:
Done ✅